src/component/RegistrationForm/RegistrationForm.tsx

```diff
@@ -0,0 +1,125 @@
+            } else if (pair[0] !== 'file') {
+                if (pair[0] === 'email') {
+                    objFormData.email = pair[1] as string;
+                } else if (pair[0] === 'password') {
+                    objFormData.password = pair[1] as string;
+                } else if (pair[0] === 'name') {
+                    objFormData.name = pair[1] as string;
+                }
+            }
+        }
+        if (flagError) {
+            onErrorSet('Empty');
+        } else {
+            if (isSignIn) {
+                onRegistrationSend(objFormData, true);
+            } else {
+                onLoginGet(objFormData, true);
+            }
+        }
+    }, [onLoginGet, onRegistrationSend, onErrorSet, isSignIn]);
+
+    useEffect(() => {
+        if (errorType === 'Not found') {
+            setErrorDescription(languages.errorSpanFound[language]);
+        } else if (errorType === 'Empty') {
+            setErrorDescription(languages.errorSpanEmpty[language]);
+        } else if (errorType === 'Exist') {
+            setErrorDescription(languages.errorSpanExist[language]);
+        } else if (errorType === 'Success') {
+            onErrorSet('');
+            history.push('/');
+        }
+    }, [language, onErrorSet, history, errorType]);
+
+    return (
+        <div className={styles.formContainer}>
+            <form className={styles.form} onSubmit={ onSubmitClick } method="POST">
+                <input className={styles.formElement} type="email" name="email" placeholder={ language === 'ru' ? 'Введите вашу почту' : language === 'en' ? 'Enter your email' : 'Entrer votre email' } />
+                <input className={styles.formElement} type="password" name="password" placeholder={ language === 'ru' ? 'Введите ваш пароль' : language === 'en' ? 'Enter your password' : 'Tapez votre mot de passe' } />
+                {
+                    isSignIn ?
+                        <>
+                            <input className={styles.formElement} type="text" name="name" placeholder={language === 'ru' ? 'Введите ваше имя' : language === 'en' ? 'Enter your name' : 'Entrez votre nom'} />
+                            <label onChange={getFile} className={styles.labelElement} >
+                                <span className={styles.fileLoad}>{ language === 'ru' ? 'Загрузите фото' : language === 'en' ? 'Upload a photo' : 'Télécharger une photo' }</span>
+                                <span>{fileInformation}</span>
+                                <input style={{ display: 'none', outline: 'none', userSelect: 'none' }} name="file" type="file" accept="image/jpeg,image/png" />
+                            </label>
+                        </>
+                        : null
+                }
+                <span>{errorDescription}</span>
+                <input className={styles.submitBtn} type="submit" value={language === 'ru' ? 'Отправить' : language === 'en' ? 'Send' : 'Envoyer'} />
+            </form>
+        </div>
```

Comment:
tabs as 2 spaces and tabs as 4 spaces is something you should keep among all files.